source/fundamentals/object-encodings.txt

```diff
@@ -0,0 +1,98 @@
+===============
+Object Encoding
+===============
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+You can customize the class that represents BSON documents in a collection
+using an optional second parameter of the
+:java-sync-api:`MongoDatabase.getCollection()
+<com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`
+method.
+
+To use a class in this way, a :java-docs:`Codec
+<apidocs/bson/org/bson/codecs/Codec.html>` for it must be registered in the
+:java-docs:`CodecRegistry
+<apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+of the ``MongoCollection``.
+
+Default Codecs
+--------------
+
+By default, every ``MongoCollection`` has ``Codec``s registered for
+three classes:
+
+- ``Document``
+- ``BasicDBObject``
+- ``BsonDocument``
+
+Custom Codecs
+-------------
+
+Applications are free to register ``Codec`` implementations for other
+classes by adding them to the ``CodecRegistry``. You can configure
+``CodecRegistry`` instances at the ``MongoClient``, ``MongoDatabase``,
+and ``MongoCollection`` levels.
+
+To register a ``Codec`` for a ``MongoClient``, use the
+:java-api-sync:`MongoClientSettings
+<com/mongodb/MongoClientSettings.html>` class:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoClientSettings settings = MongoClientSettings.builder()
+        .codecRegistry(codecRegistry).build();
+   MongoClient client = MongoClients.create(settings);
+
+To register a ``Codec`` for a ``MongoDatabase``, use the
```

Comment:
Since there aren't really significant differences between the three, maybe there's a better way to show this without repeating the code to construct codecRegistry?